### PR TITLE
Update SDL analyzers to latest

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Also remove some deprecated analyzers

See also: https://github.com/Azure/azure-iot-sdk-csharp/pull/3206/

### CODE REVIEW

File: vsts/sdl.yaml [hunk 1]
```diff
@@ -81,24 +69,24 @@ jobs:
...
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-uploadtotsa.TSAUpload@2
     displayName: 'TSA upload'
     inputs:
       tsaVersion: TsaV2
       codebase: NewOrUpdate
       tsaEnvironment: PROD
       codeBaseName: 'Azure-Iot-SDK-Java-Master'
       notificationAlias: 'aziotdndmgdsdk@microsoft.com'
-      codeBaseAdmins: 'REDMOND\timtay;REDMOND\jamdavi;REDMOND\azabbasi'
+      codeBaseAdmins: 'REDMOND\timtay;'
```
Comment (drwill-ms) [inline vsts/sdl.yaml:89]: Add Andy

File: vsts/sdl.yaml [hunk 2]
```diff
@@ -81,24 +69,24 @@ jobs:
...
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@2
     displayName: 'Post Analysis'
     inputs:
       CredScan: true
       Semmle: true
 
-  - task: securedevelopmentteam.vss-secure-development-tools.build-task-uploadtotsa.TSAUpload@1
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-uploadtotsa.TSAUpload@2
     displayName: 'TSA upload'
     inputs:
```
Comment (drwill-ms) [inline vsts/sdl.yaml:83]: These will change to use a file and only have 3 (new) inputs.

File: vsts/sdl.yaml [hunk 3]
```diff
@@ -25,24 +25,12 @@ jobs:
       alertWarningLevel: High
       failOnAlert: true
 
-  - task: securedevelopmentteam.vss-secure-development-tools.build-task-credscan.CredScan@2
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-credscan.CredScan@3
     displayName: CredScan
     inputs:
       debugMode: false
       suppressionsFile: vsts/CredScanSuppressions.json
-      toolMajorVersion: V2
```
[the remark this answers is not in the document]
Comment (timtay-microsoft) [reply vsts/sdl.yaml:33]: I'm not positive, but it seems to work fine

[the remark this answers is not in the document]
Comment (timtay-microsoft) [reply vsts/sdl.yaml:33]: I'll try removing this field completely and see if it was even being referenced

**Review by drwill-ms — APPROVED** [review]

File: vsts/sdl.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (timtay-microsoft) [reply vsts/sdl.yaml:33]: Removing it seems to have no impact, so you can remove it on the .NET side as well

**Pull request merged**